src/state/StateKeyValue.cpp

```diff
@@ -419,21 +419,23 @@ namespace state {
         pushPartialToRemote(dirtyMaskBytes);
     }
 
-    long StateKeyValue::waitOnRedisRemoteLock(const std::string &redisKey) {
+    std::optional<long> StateKeyValue::waitOnRedisRemoteLock(const std::string &redisKey) {
         PROF_START(remoteLock)
 
-        long remoteLockId = redis.acquireLock(redisKey, REMOTE_LOCK_TIMEOUT_SECS);
+        std::optional<long> remoteLockId = redis.acquireLock(redisKey, REMOTE_LOCK_TIMEOUT_SECS);
         unsigned int retryCount = 0;
-        while (remoteLockId <= 0) {
-            logger->debug("Waiting on remote lock for {} (loop {})", redisKey, retryCount);
+
+        while (!remoteLockId.has_value()) {
+            logger->info("Waiting on remote lock for {} (loop {})", redisKey, retryCount);
 
             if (retryCount >= REMOTE_LOCK_MAX_RETRIES) {
                 logger->error("Timed out waiting for lock on {}", redisKey);
                 break;
             }
 
-            // Sleep for 1ms
-            usleep(1000);
+            // Sleep for 1ms + noise
+            std::chrono::microseconds ms(500 + (rand() % 3000));
```

Comment:
Very tiny comment but given microseconds I'd call the variable `us` or `micros`. I've seen too many issues confusing the two before which is why I'm hyper sensitive to it

Reply:
I'd even prefer using the name constructors like `1ms` but it doesn't work 😕.
I oscillated in my tests between using `milliseconds`, `microseconds` and `nanoseconds` so it's actually a very good catch!

Reply:
Actually, make sure you turn on unicode support, i'm going to call it `μs` 🙃

Reply:
I removed it entirely. randomisation of wait time isn't a great way to do things and may hide issues we'd wanna spot